src/lib/y2partitioner/actions/controllers/filesystem.rb

```diff
@@ -228,25 +227,84 @@ def partition_id=(partition_id)
           blk_device.adapted_id = partition_id
         end
 
-        # Sets the mount point of the filesystem if there is one
+        # Creates a mount point for the current filesystem
         #
-        # Take into account that modifying the mount point can have side effects
-        # if the filesystem doesn't exist yet, like changing the list of
-        # subvolumes if the new or old mount point is "/".
+        # @note The new mount point is created with default mount options if no mount
+        #   options are given (see {Y2Storage::Filesystems::Type#default_mount_options}).
+        #
+        #   Take into account that modifying the mount point can have side effects
+        #   if the filesystem doesn't exist yet, like changing the list of
+        #   subvolumes if the new or old mount point is "/".
+        #
+        # @param path [String]
+        # @param options [Hash] options for the mount point (e.g., { mount_by: :uuid } )
+        def create_mount_point(path, options = {})
+          return if filesystem.nil? || !mount_point.nil?
```

Comment:
IMHO this is a bit hard to read: It checks for two completely different things:
- Do we already have a filesystem? Return if we don't.
- Do we already have a mount point? Return if we do.
I know that it is correct in the logical sense, but you have to do mental acrobatics to figure out exactly when this returns.

Reply:
Totally agree. I will split it.